libs/fetcher.py

```diff
@@ -0,0 +1,347 @@
+        # 最高价计算及赋值
+        # EURUSD和GBPUSD为反向取值，所以取dxy的最高值的时候需要取这两个货币对的最低值，其余正相关货币对取最高值
+        if symbol_name == 'EURUSD' or symbol_name == 'GBPUSD':
+            dxy_rate['price_high'] = math.pow(symbol_low_price, dxy_portions[symbol_name]) * dxy_rate['price_high']
+        else:
+            dxy_rate['price_high'] = math.pow(symbol_high_price, dxy_portions[symbol_name]) * dxy_rate['price_high']
+        # 最低价计算及赋值
+        # EURUSD和GBPUSD为反向取值，所以取dxy的最低值的时候需要取这两个货币对的最高值
+        if symbol_name == 'EURUSD' or symbol_name == 'GBPUSD':
+            dxy_rate['price_low'] = math.pow(symbol_high_price, dxy_portions[symbol_name]) * dxy_rate['price_low']
+        else:
+            dxy_rate['price_low'] = math.pow(symbol_low_price, dxy_portions[symbol_name]) * dxy_rate['price_low']
+        # 收盘价计算及赋值
+        dxy_rate['price_closed'] = math.pow(symbol_closed_price, dxy_portions[symbol_name]) * dxy_rate['price_closed']
+    # 返回结果列表
+    return dxy_rate
+
+
+# 根据mt5数据，计算该时间区间内的DXY报价
+def get_dxy_from_mt5(start, end, interval):
+    # start和end必须为utc时间(mt5默认就是使用的utc时间)
+    # 定义组成DXY的6个货币对名称，由于拉取数据时所用的标准化名称与实际名称一致并且没有特殊字符，所以这里就不去symbol method表中查，直接赋值即可
+    dxy_symbol_list = ('EURUSD', 'USDJPY', 'GBPUSD', 'USDCAD', 'USDSEK', 'USDCHF')
+    # 根据interval选择对应的处理逻辑，一切以最小粒度分钟为基础进行组合。
+    result_list = []
+
+    # 目前仅支持1m和1h，如果有需要就继续添加if分支即可
+    # interval大于1分钟的最低价和最高价均不准确，详情请参考：https://trello.com/c/Eqf2N3m5
+    if interval == '1m':
+        # 将起始时间秒强制取0,向下取整
+        start_rounding_time = start.replace(second=0, microsecond=0)
+        end_rounding_time = end.replace(second=0, microsecond=0)
+        # 获取起始时间的分钟差值: end要比start大至少1分钟
+        delta_minute = int((end_rounding_time.timestamp() - start_rounding_time.timestamp()) / 60)
+        if delta_minute < 0:
+            logger.error("结束时间(%s)要大于等于开始时间(%s)1分钟，请检查输入参数是否正确!", end, start)
+            return []
+        else:
+            # 拉取开始时间这个时间戳的数据，开始和结束时间一致即可
+            for minute in range(0, delta_minute + 1):
+                # 清空这上一个周期内的symbols_rate_list
+                symbols_rate_list = []
+                # 重置skip值，避免只要有一个货币对在一个时间点出现问题就一直跳过
+                skip = False
+                # 遍历dxy_symbol_list,拉取这个时间周期的所有symbol报价,用于计算dxy
+                the_time_by_minute = start_rounding_time + timedelta(minutes=minute)
+                for symbol in dxy_symbol_list:
+                    symbol_rates = get_historical_data_from_mt5(symbol, interval,
+                                                                the_time_by_minute, the_time_by_minute)
+                    # 发现一个新问题，mt5中某个时间点的数据有可能为空。碰到这种情况就直接跳过即可
+                    if len(symbol_rates) > 0:
+                        symbols_rate_list.append(symbol_rates[0])
+                    else:
+                        logger.error('%s在%s的数据缺失! @interval: %s', symbol, str(the_time_by_minute), interval)
+                        skip = True
+                        break
+
+                # 只要有一个货币对在这一小时的数据为空，则跳过这一分钟的计算
```

Comment:
```suggestion
                # 只要有一个货币对在这一分钟的数据为空，则跳过这一分钟的计算
```
并且这个注释应该往上挪到skip = True那